fe/fe-core/src/main/java/com/starrocks/catalog/LocalTablet.java

```diff
@@ -67,8 +68,15 @@ public enum TabletStatus {
         NEED_FURTHER_REPAIR, // one of replicas need a definite repair.
     }
 
+    // Most read only accesses to replicas should acquire db lock, to prevent
+    // modification to replicas list during read access.
+    // This method avoids acquiring db lock by acquiring replicas object lock
+    // instead, so the lock granularity is reduced.
+    // To achieve this goal, all write operations to replicas object should
```

Comment:
no, just write operation

Reply:
other read operation is protected by db lock

Reply:
So, when it's read operation, if we don't want to use `synchronized (replicas)`, we need to acquire some kind of higher level lock, like DB lock? i am a little confused here, because the following function is justing doing read op here but using synchronized,

```
    // Note: this method does not require db lock to be held
    public boolean quorumReachVersion(long version, long quorum) {
        long valid = 0;
        synchronized (replicas) {
            for (Replica replica : replicas) {
                if (replica.checkVersionCatchUp(version, true)) {
                    valid++;
                }
            }
        }
        return valid >= quorum;
    }
```

Reply:
> other read operation is protected by db lock

got it, i think we'd better specify this in the comments.